core/src/components/textarea/textarea.scss

```diff
@@ -283,3 +304,219 @@
+   */
+  @include margin(4px, 0, 0, 0);
+}
+
+:host(.textarea-label-placement-start) .textarea-wrapper textarea,
+:host(.textarea-label-placement-end) .textarea-wrapper textarea,
+:host(.textarea-label-placement-fixed) .textarea-wrapper textarea {
+  @include padding(10px, 0, 0, 0);
+}
+
+// Textarea Label Placement - Start
+// ----------------------------------------------------------------
+
+/**
+ * Label is on the left of the textarea in LTR and
+ * on the right in RTL.
+ */
+:host(.textarea-label-placement-start) .textarea-wrapper {
+  flex-direction: row;
+}
+
+:host(.textarea-label-placement-start) .label-text-wrapper {
+  /**
+   * The margin between the label and
+   * the textarea should be on the end
+   * when the label sits at the start.
+   */
+  @include margin(0, 8px, 0, 0);
+}
+
+// Textarea Label Placement - End
+// ----------------------------------------------------------------
+
+/**
+ * Label is on the right of the textarea in LTR and
+ * on the left in RTL.
+ */
+:host(.textarea-label-placement-end) .textarea-wrapper {
+  flex-direction: row-reverse;
+}
+
+/**
+ * The margin between the label and
+ * the textarea should be on the start
+ * when the label sits at the end.
+ */
+:host(.textarea-label-placement-end) .label-text-wrapper {
+  @include margin(0, 0, 0, 8px);
+}
+
+// Textarea Label Placement - Fixed
+// ----------------------------------------------------------------
+
+:host(.textarea-label-placement-fixed) .label-text-wrapper {
+  /**
+   * The margin between the label and
+   * the textarea should be on the end
+   * when the label sits at the start.
+   */
+  @include margin(0, 8px, 0, 0);
+}
+
+/**
+ * Label is on the left of the textarea in LTR and
+ * on the right in RTL. Label also has a fixed width.
+ */
+:host(.textarea-label-placement-fixed) .label-text {
+  flex: 0 0 100px;
+
+  width: 100px;
+  min-width: 100px;
+  max-width: 200px;
+}
+
+// Textarea Label Placement - Stacked and Floating
+// ----------------------------------------------------------------
+
+/**
+ * Stacked: Label sits above the textarea and is scaled down.
+ * Floating: Label sits over the textarea when the textarea has no
+ * value and is blurred. Label sits above the textarea and is scaled
+ * down when the textarea is focused or has a value.
+ *
+ */
+:host(.textarea-label-placement-stacked) .textarea-wrapper,
+:host(.textarea-label-placement-floating) .textarea-wrapper {
+  flex-direction: column;
+  align-items: start;
+}
+
+/**
+ * Ensures that the label animates
+ * up and to the left in LTR or
+ * up and to the right in RTL.
+ */
+:host(.textarea-label-placement-stacked) .label-text-wrapper,
+:host(.textarea-label-placement-floating) .label-text-wrapper {
+  @include transform-origin(start, top);
+
+  max-width: 100%;
+}
+
+/**
+ * This makes the label sit over the textarea
+ * when the textarea is blurred and has no value.
+ */
+:host(.textarea-label-placement-floating) .label-text-wrapper {
```

Comment:
The label alignment is odd when the label overlays the textarea:
![image](https://user-images.githubusercontent.com/2721089/206014332-061fb252-d01f-4025-9793-b0841ccee693.png)


I realize it is not supposed to be centered according to https://material-components.github.io/material-components-web-catalog/#/component/text-field

When comparing with the textareas in the link above, it looks like the text is positioned slightly higher than where the label sat (where this PR has it the other way around)